gobblin-service/src/main/java/org/apache/gobblin/service/modules/scheduler/GobblinServiceJobScheduler.java

```diff
@@ -332,10 +332,13 @@ public AddSpecResponse onAddSpec(Spec addedSpec) {
     // Check quota limits against run immediately flows or adhoc flows before saving the schedule
     // In warm standby mode, this quota check will happen on restli API layer when we accept the flow
     if (!this.warmStandbyEnabled && (!jobConfig.containsKey(ConfigurationKeys.JOB_SCHEDULE_KEY) || PropertiesUtils.getPropAsBoolean(jobConfig, ConfigurationKeys.FLOW_RUN_IMMEDIATELY, "false"))) {
+      // This block should be reachable only for the first execution for the adhoc flows (flows that either do not have a schedule or have runImmediately=true.
```

Comment:
Just a reminder: As on line 334 we checked whether  this.warmStandbyEnabled should  set to be false, this will only be reached in the old mode, where we didn't enable message forwarding. 
You should check quota on the compiler onAddSpec method when we enable message forwarding